### PR TITLE
[Dashlane Vault] Master password preferences and improved error handling

### LINKED ISSUE #11715: [Dashlane Vault] Empty Vault
### Extension

https://www.raycast.com/tm.wrnr/dashlane-vault

### Description

The Extension I properly installed and also my Path is updated in the extension itself to the Path where the dcli is stored. But still I get none of my Password inside of the Extension. I just get "Vault Empty" and when I sync I get "Sync with Dashlane succeeded".  I CAN however access my Notes those are synced Properly but I cannot access any of the Passwords inside of the Vault.

### Steps To Reproduce

1. Install Homebrew
2. Install the DCLI
3. Login with the DCLI and Sync
4. Install the Raycast Extension
5. Change the Path to the DCLI in the Extension
6. Don't get any Password inside of the Vault

### Current Behaviour

No Passwords shown inside of the Vault

### Expected Behaviour

All my Saved passwords inside of my Vault

### PR DESCRIPTION
**Pull request by tmwrnr**
### Master Password Preferences

Users can now save their Dashlane CLI master password in the Raycast extension preferences. This is useful if the user uses the following CLI configuration:

```sh
dcli c save-master-password false
```

With this config the user needs to insert the master password every time (related to #11326). With this new feature the extension user can use this configuration and still use the extension without entering the password every time.

### Biometrics Preferences

User can now use a Dashlane CLI configuration to unlock the vault with biometrics. 

### Error Handling

Currently the user does not know if something wrong happens within the extension. The fallback was to show an empty view. With the new changes detailed error messages with the help of `showFailureToast` from `@raycast/utils`.

The user is also shown an error message if the user has the CLI version [6.2415.0](https://github.com/Dashlane/dashlane-cli/releases/tag/v6.2415.0) installed, where a bug was introduced. Fixes #11715

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/dashlane-vault/README.md [hunk 1]
```diff
@@ -29,8 +29,32 @@ brew install dashlane/tap/dashlane-cli
...
+
+Retrieve the path to the Dashlane CLI by running the following command in your terminal:
+
+```sh
+which dcli
+```
+
+Set the path to the Dashlane CLI in the extension preferences of Raycast. Open Raycast preferences, navigate to the Dashlane Vault extension, and specify the path to the Dashlane CLI. You can also use the following shortcut to jump directyl to the extenions settings: `⌥ + ⌘ + ,`
+
+### Master Password
```
Comment (Mikescops) [inline extensions/dashlane-vault/README.md:50]: Could add a short explanation on how to use biometrics unlock?

File: extensions/dashlane-vault/src/helper/error.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export function getErrorAction(error: unknown) {
+  if (error instanceof DisplayableError) return error.action;
+}
+
+export function getDisplayableErrorMessage(error: unknown) {
+  if (error instanceof DisplayableError) return error.message;
+  return undefined;
+}
+
+export const getErrorString = (error: unknown): string | undefined => {
```
Comment (Mikescops) [inline extensions/dashlane-vault/src/helper/error.ts:55]: Curious that raycast doesn't have this kind of helpers in their core library

**Review by Mikescops — CHANGES_REQUESTED** [review]
Thanks for this 👍

File: extensions/dashlane-vault/README.md (hunk 1 above)
Comment (tmwrnr) [reply extensions/dashlane-vault/README.md:50]: Currently this pull request does not introduce this feature to the extension yet. I tried it and it mostly worked. 

But I noticed that if you set `save-master-password` to false and `user-presence` to biometric you can no longer use the master password within the environment variables. So you will have to insert the password after the biometric successfully was requested, which is not possible by using it via Nodejs. Could you confirm this @Mikescops?

But I will add the possibility to use either save-master-password or biometrics within this pull request.

File: extensions/dashlane-vault/src/helper/error.ts (hunk 2 above)
Comment (tmwrnr) [reply extensions/dashlane-vault/src/helper/error.ts:55]: No I double checked it.

Comment (pernielsentikaer) [reply extensions/dashlane-vault/src/helper/error.ts:55]: I can't see the context on this screen, can this help? https://developers.raycast.com/utilities/functions/showfailuretoast#signature

Comment (tmwrnr) [reply extensions/dashlane-vault/src/helper/error.ts:55]: @pernielsentikaer In this context I want to pass the returned value from `getErrorString` to a custom error, which will than be passed to `showFailureToast`.

File: extensions/dashlane-vault/README.md [hunk 3]
```diff
@@ -55,14 +83,18 @@ which dcli
 
 4. Set the path to the Dashlane CLI in the extension preferences of Raycast. Open Raycast preferences, navigate to the Dashlane Vault extension, and specify the path to the Dashlane CLI.
 
+### Unsupported Versions
+
+Currently there is a version (v6.2415.0) of the Dashlane CLI with a bug, that can only be fixed by upgrading the CLI.
+
+### Wrong configuration
+
+Currently the extensions does not support using both configurations `save-master-password` set to false and `user-presence` set to biometrics. Please use only one of the configuration.
```
Comment (Mikescops) [inline extensions/dashlane-vault/README.md:92]:
```suggestion
Currently the extension does not support using both configurations `save-master-password` set to false and `user-presence` set to biometrics. Please use only one of the configuration.
```

File: extensions/dashlane-vault/README.md (hunk 1 above)
Comment (Mikescops) [reply extensions/dashlane-vault/README.md:50]: I tried `save-master-password` to false and `user-presence` to biometrics and passing the env variable `DASHLANE_MASTER_PASSWORD` works just fine for me.

Also I'm not sure to understand why you put a condition on biometrics to open the raycast app?

Comment (tmwrnr) [reply extensions/dashlane-vault/README.md:50]: > I tried save-master-password to false and user-presence to biometrics and passing the env variable DASHLANE_MASTER_PASSWORD works just fine for me.

I checked it again and it is working as you mentioned. @Mikescops 

> Also I'm not sure to understand why you put a condition on biometrics to open the raycast app?

If the biometrics prompt is shown the raycast app is not visible anymore. After the prompt was successful, I want to show the raycast app again.

Comment (Mikescops) [reply extensions/dashlane-vault/README.md:50]: Alright good for me

**Review by Mikescops — APPROVED** [review]
Looks good now

**Pull request merged**